source/agora/consensus/data/Block.d

```diff
@@ -721,16 +739,27 @@ version (unittest)
         {
             Scalar v = secretKeyToCurveScalar(key.secret);
             // rc = r used in signing the commitment
-            const Scalar rc = Scalar(hashMulti(v, "consensus.signature.noise", cycle));
+            const Scalar rc = Scalar(hashMulti(v, "consensus.signature.noise",
+                cycleForValidator(key.address)));
             const Scalar r = rc + challenge; // make it unique each challenge
             const Pair R = Pair.fromScalar(r);
             const K = Point(key.address[]);
-            sigs ~= multiSigSign(R, v, challenge);
+            auto sig = multiSigSign(R, v, challenge);
+            log.trace("multiSigTestBlock: cycle {} index {} Commited R for validator {} is \n{} \nsig is {}",
+                cycleForValidator(key.address), i, key.address, rc.toPoint(), sig);
+            sigs ~= sig;
             validators[i] = true;
         }
-        keys.enumerate.each!((idx, key) => validatorSign(idx, key));
-        block.header.signature = multiSigCombine(sigs);
+        try
+        {
+            keys.enumerate.each!((idx, key) => validatorSign(idx, key));
+        } catch (Exception e)
+        {
+            log.error("Unit test signing error: {}", e);
```

Comment:
If it's a unittest failure then it should assert, not just log. But I think the issue here is that `validatorSign` is not nothrow. Which part can throw? Could we just add an assert?

Reply:
Made it assert with the message.